bundles/org.openhab.binding.valloxmv/src/main/java/org/openhab/binding/valloxmv/internal/ValloxMVHandler.java

```diff
@@ -119,16 +119,13 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                 return;
             }
             if (strUpdateValue != "") {
-                // Open WebSocket
-                valloxSendSocket.request(channelUID, strUpdateValue);
-                // We should wait some time to let device process the data
-                try {
-                    Thread.sleep(500);
-                } catch (InterruptedException e) {
-                    // sleep interrupted, we will get updated values during next data update cycle
-                    return;
-                }
-                valloxSendSocket.request(null, null);
+                // Stop scheduler while sending command to device
+                dispose();
```

Comment:
I think calling dispose right here is inappropriate. Sure, right now all it does is cancels the update tasks but `dispose` could eventually do far more than that, and it is misleading reading the code since it appears you are disposing the entire handler every time you process a command. Refactor the code in your dispose method into a separate method called `cancelUpdateTask` or `disposeUpdateTask` or any other name of your choosing then call that method instead.

Reply:
Fixed. Renamed job to what it really is. Added more logic using next read data time, that is not 100% accurate but it does not need to be - logic is based on seconds and not milliseconds. Fixed refresh request - with original code it was never run.